backend/dal/Configuration/ProjectNumberConfiguration.cs

```diff
@@ -0,0 +1,25 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using Pims.Dal.Entities;
+
+namespace Pims.Dal.Configuration
+{
+    /// <summary>
+    /// ProjectNumberConfiguration class, provides a way to configure project numbers in the database.
+    ///</summary>
+    public class ProjectNumberConfiguration : BaseEntityConfiguration<ProjectNumber>
```

Comment:
We use the `ProjectNumber` to provide a way to generate unique numbers for **Projects**.  The reason we don't simply have **Project.Id** is because we wouldn't be able to at the same time generate the **Project.ProjectNumber** (which is the actual primary key and unique value to identify the project.

The downside is we can generate project numbers that get orphaned under some conditions.